src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,41 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can not add country: " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
Consider using try-with-resources for the `Session` object in the `add` method. This will ensure that the session is closed automatically, even if an exception occurs, and will help prevent resource leaks.